### PR TITLE
DAOS-12221 test: Update implementation of dmg stop, dmg start in test code.

### PR DESCRIPTION
**Pull request by jgmoore-or**
Some tests (e.g., daos_test - rebuild_ec) currently might fail due the invocation of dmg stop failing to stop the specified rank.  In these cases, the invocation ( via system() ) returns "success" but the failure to stop the rank causes the overall test to fail.

This PR replaces the dmg calls via system() with the daos_dmg_json_pipe() function implemented within the DAOS tests helpers module.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/suite/daos_test_common.c [hunk 1]
```diff
@@ -921,8 +921,8 @@ daos_start_server(test_arg_t *arg, const uuid_t pool_uuid,
 	if (arg->dmg_config != NULL)
 		dts_append_config(dmg_cmd, " -o %s", arg->dmg_config);
 
-	rc = system(dmg_cmd);
-	print_message(" %s rc %#x\n", dmg_cmd, rc);
+	rc = dmg_system_start_rank(dmg_config_file, rank, arg->dmg_config);
```
Comment (mjmac) [inline src/tests/suite/daos_test_common.c:924]: Why are you passing in the config file twice?

File: src/common/tests_dmg_helpers.c [hunk 2]
```diff
@@ -1221,6 +1221,67 @@ int verify_blobstore_state(int state, const char *state_str)
...
+	int			argcount = 0;
+	char			**args = NULL;
+	struct json_object	*dmg_out = NULL;
+	int			rc = 0;
+
+	args = cmd_push_arg(args, &argcount, " -r %d ", rank);
+	if (args == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	args = cmd_push_arg(args, &argcount, "-o ");
```
Comment (mjmac) [inline src/common/tests_dmg_helpers.c:1265]: Why are you handling this here? In `daos_dmg_json_pipe()`, this is handled for you if `dmg_config_file` is not NULL: https://github.com/daos-stack/daos/pull/11015/files#diff-0c458960127d7ac3a0091dbaa481a576ea6d3cf0d231c6e2bc2bc1672892e332R257

**Review by kjacque — APPROVED** [review]

File: src/include/daos/tests_lib.h [hunk 3]
```diff
@@ -361,6 +361,26 @@ int dmg_storage_query_device_health(const char *dmg_config_file, char *host,
...
+ * \param force	[IN]	Terminate with extreme prejudice.
+ */
+int dmg_system_stop_rank(const char *dmg_config_file, d_rank_t rank, int force);
+
+/**
+ * Start a rank.
+ *
+ * \param dmg_config_file
+ *		[IN]	DMG config file
+ * \param rank	[IN]	Rank to stop.
```
Comment (frostedcmos) [inline src/include/daos/tests_lib.h:379]: should be 'Rank to start' in the comment

**Review by frostedcmos — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

File: src/tests/suite/daos_test_common.c [hunk 4]
```diff
@@ -967,13 +960,10 @@ daos_kill_server(test_arg_t *arg, const uuid_t pool_uuid,
 		      "disabled, svc->rl_nr %d)!\n", rank, arg->srv_ntgts,
 		       arg->srv_disabled_ntgts - 1, svc->rl_nr);
 
-	/* build and invoke dmg cmd to stop the server */
-	dts_create_config(dmg_cmd, "dmg system stop -r %d --force", rank);
-	if (arg->dmg_config != NULL)
-		dts_append_config(dmg_cmd, " -o %s", arg->dmg_config);
+	/* stop the rank */
+	rc = dmg_system_stop_rank(dmg_config_file, rank, true);
+	print_message("dmg stop, rc %#x\n", rc);
```
[the remark this answers is not in the document]
Comment (jgmoore-or) [reply src/tests/suite/daos_test_common.c]: fixed.

**Pull request merged**